### PR TITLE
Fix vega warnings 2

### PR DESCRIPTION
**Pull request by b-yogesh**
This PR fixes the previous PR https://github.com/bcdev/chartlets/pull/37 which had build errors and was then reverted.
What has changed since that PR is 
- Removed source exports
- Modified the `signals` const that creates the map of listeners and their names.
- Version bumped to `0.0.29` for both frontend and backend

### CODE REVIEW

File: chartlets.js/src/lib/hooks.ts [hunk 1]
```diff
@@ -38,6 +42,94 @@ export function makeContributionsHook<S extends object = object>(
...
+          isString(param.select.on),
+      )
+      .reduce(
+        (acc, param) => {
+          acc[param.select.on] = param.name;
+          return acc;
+        },
+        {} as { [key: string]: string },
+      );
+  }, [chart]);
```
Comment (b-yogesh) [inline chartlets.js/src/lib/hooks.ts:78]: This is the part that was changed

File: chartlets.py/my_extension/my_panel_1.py [hunk 2]
```diff
@@ -79,5 +83,45 @@ def make_figure(ctx: Context, selected_dataset: int = 0) -> alt.Chart:
...
         .add_params(corner_var, click_param)
     )
-
     return chart
+
+
+@panel.callback(
+    Input("plot", property="points"), State("plot", "chart"), Output("plot", "chart")
+)
+def get_click_event_points(ctx: Context, points, plot) -> alt.Chart:
```
Comment (forman) [inline chartlets.py/my_extension/my_panel_1.py:92]: This is not the intended way to implement a callback.

- The input should not be the hole Vega chart object. They can be very large (data!) and need to be transported from server to UI on evary callback!
- The output in this case should also not b a chart, but a property within the chart, here `Output("plot", "chart.encoding.color")`
- Name your arguments according to the properties, not `plot`, but `chart`
- Add type annotations if you know the types

**Review by forman — CHANGES_REQUESTED** [review]
Also update CHANGES.md and increase version numbers

File: chartlets.py/my_extension/my_panel_1.py (hunk 2 above)
Comment (b-yogesh) [reply chartlets.py/my_extension/my_panel_1.py]: Thanks for the feedback. Have done the changes.

**Review by forman — APPROVED** [review]

**Pull request merged**